### PR TITLE
feat: create EZA_ICONS_AUTO environment variable

### PR DESCRIPTION
**Pull request by RyanBreaker**
Adds support for a new `EZA_ICONS_AUTO` environment variable that, if set, automates the same behavior as using `--icons` or `--icons=auto`. Any explicit use of `--icons=WHEN` overrides this behavior.

This is just a feature I thought I'd try implementing for my own use because I like having icons always on and figured it could be useful for others. This is also my first ever open source Rust project contribution so please feel free to give feedback!

### CODE REVIEW

File: man/eza.1.md [hunk 1]
```diff
@@ -302,6 +302,12 @@ For more information on the format of these environment variables, see the [eza_
 
 Overrides any `--git` or `--git-repos` argument
 
+## `EZA_ICONS_AUTO`
+
+If set to `true`, automates the same behavior as `--icons` or `--icons=auto`. Useful for if you always want to have icons enabled.
```
Comment (MartinFillon) [inline man/eza.1.md:307]: We could just have it setted and not only true

File: src/options/vars.rs [hunk 2]
```diff
@@ -55,6 +55,8 @@ pub static EZA_ICON_SPACING: &str = "EZA_ICON_SPACING";
 pub static EXA_OVERRIDE_GIT: &str = "EXA_OVERRIDE_GIT";
 pub static EZA_OVERRIDE_GIT: &str = "EZA_OVERRIDE_GIT";
 
+pub static EZA_ICONS_AUTO: &str = "EZA_ICONS_AUTO";
```
Comment (MartinFillon) [inline src/options/vars.rs:58]: Please add a description as for all the other variables

**Review by MartinFillon — CHANGES_REQUESTED** [review]

File: man/eza.1.md (hunk 1 above)
Comment (RyanBreaker) [reply man/eza.1.md]: Good point, changed to this in the latest commit.

**Review by cafkafk — APPROVED** [review]
Integration tests and my local testing of this having the correct behavior and man-page output indicates this as working, thanks for the contribution!

**Review by MartinFillon — APPROVED** [review]

**Comment by MartinFillon** [comment]
This is great merging after main merging done if tests passes

**Pull request merged**